pkg/backend/filestate/bucket_test.go

```diff
@@ -0,0 +1,94 @@
+package filestate
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"gocloud.dev/blob"
+)
+
+func mustNotHaveError(t *testing.T, context string, err error) {
+	t.Helper()
+	if err != nil {
+		t.Fatalf("Error in testcase %q, aborting: %v", context, err)
+	}
+}
+
+// The wrappedBucket type exists so that when we use the blob.Bucket type we can present a consistent
+// view of file paths. Since it will assume that backslashes (file separators on Windows) are part of
+// file names, and this causes "problems".
+func TestWrappedBucket(t *testing.T) {
+	// wrappedBucket will only massage file paths IFF it is needed, as filepath.ToSlash is a noop.
+	if filepath.Separator == '/' {
+		assert.Equal(t, `foo\bar\baz`, filepath.ToSlash(`foo\bar\baz`))
+		t.Skip("Skipping wrappedBucket tests because file paths won't be modified.")
+	}
+
+	// Initialize a filestate backend, using the default Pulumi directory.
+	cloudURL := FilePathPrefix + "~"
+	b, err := New(nil, cloudURL)
+	if err != nil {
+		t.Fatalf("Initializing new filestate backend: %v", err)
+	}
+	localBackend, ok := b.(*localBackend)
+	if !ok {
+		t.Fatalf("backend wasn't of type localBackend?")
+	}
+
+	wrappedBucket, ok := localBackend.bucket.(*wrappedBucket)
+	if !ok {
+		t.Fatalf("localBackend.bucket wasn't of type wrappedBucket?")
+	}
+
+	ctx := context.Background()
+	// Perform basic file operations using wrappedBucket and verify that it will
+	// successfully handle both "/" and "\" as file separators. (And probably fail in
+	// exciting ways if you try to give it a file on a system that supports "\" or "/" as
+	// a valid character in a filename.)
```

Comment:
Are there systems that do that? Just curious...

Reply:
Not to my knowledge.